src/main/java/org/folio/circulation/domain/OverdueFineCalculatorService.java

```diff
@@ -101,20 +103,30 @@ private OverdueFineCalculatorService(Clients clients) {
 
     OverdueFinePolicy overdueFinePolicy = loan.getOverdueFinePolicy();
     if (overdueMinutes > 0 && overdueFinePolicy != null) {
-      Double maxFine = loan.wasDueDateChangedByRecall() ?
-        overdueFinePolicy.getMaxOverdueRecallFine() :
-        overdueFinePolicy.getMaxOverdueFine();
-
-      OverdueFineInterval interval = overdueFinePolicy.getOverdueFineInterval();
-      Double overdueFinePerInterval = loan.getOverdueFinePolicy().getOverdueFine();
-
-      if (maxFine != null && interval != null && overdueFinePerInterval != null) {
-        double numberOfIntervals = Math.ceil(overdueMinutes.doubleValue() /
-          interval.getMinutes().doubleValue());
-        overdueFine = overdueFinePerInterval * numberOfIntervals;
-
-        if (maxFine > 0) {
-          overdueFine = Math.min(overdueFine, maxFine);
+      OverdueFinePolicyFineInfo fineInfo = overdueFinePolicy.getFineInfo();
+      OverdueFinePolicyLimitInfo limitInfo = overdueFinePolicy.getLimitInfo();
+      if (fineInfo != null && limitInfo != null) {
+        Double finePerInterval = loan.wasDueDateChangedByRecall() ?
+          fineInfo.getOverdueRecallFine() :
+          fineInfo.getOverdueFine();
+
+        OverdueFineInterval interval = loan.wasDueDateChangedByRecall() ?
+          fineInfo.getOverdueRecallFineInterval() :
+          fineInfo.getOverdueFineInterval();
```

Comment:
Could this logic be better encapsulated by having a single branching statement that provides all of the varying parameters, e.g. the fine (amount / rate ?) and interval.

Ideally, in the future maybe, the parameters could be constructed separately to the application meaning there was only one place where the distinction between return and renewal is made.

Does that make sense?

Reply:
> Could this logic be better encapsulated by having a single branching statement that provides all of the varying parameters, e.g. the fine (amount / rate ?) and interval.
> 
> Maybe this logic could be encapsulated in the overdue fine policy?

Moved to OverdueFinePolicy

> Ideally, in the future maybe, the parameters could be constructed separately to the application meaning there was only one place where the distinction between return and renewal is made.

The distinction here is between recall and non-recall cases (whether due date has been changed by a recall request), it'll work for both renew and chek-in actions.